p2p/transport/webrtc/stream_write.go

```diff
@@ -0,0 +1,204 @@
+package libp2pwebrtc
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/libp2p/go-libp2p/p2p/transport/webrtc/internal/async"
+	pb "github.com/libp2p/go-libp2p/p2p/transport/webrtc/pb"
+	"github.com/libp2p/go-msgio/pbio"
+	"github.com/pion/datachannel"
+	"google.golang.org/protobuf/proto"
+)
+
+// Package pion detached data channel into a net.Conn
+// and then a network.MuxedStream
+type (
+	webRTCStreamWriter struct {
+		stream *webRTCStream
+
+		writer    pbio.Writer
+		writerMux sync.Mutex
```

Comment:
I don't think we need this mutex, it's invalid to call `Write` concurrently.

Reply:
This was to protect the user, but if we can make this assumption as part of the transport contract, fair enough for me.